crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -293,6 +293,27 @@ impl AuthorityPerEpochStore {
         db_options: Option<Options>,
         metrics: Arc<EpochMetrics>,
         epoch_start_configuration: Option<EpochStartConfiguration>,
+    ) -> Arc<Self> {
+        Self::new_internal(
+            name,
+            committee,
+            parent_path,
+            db_options,
+            metrics,
+            epoch_start_configuration,
+            None,
+        )
+    }
+
+    // Internal new that takes an optional Protocol config
+    fn new_internal(
```

Comment:
I made `new` be `new_internal` taking a `ProtocolConfig` and being called from `new_for_testing`.
`new` stays as is and should keep everything as it was